lib/i3ipc/protocol.rb

```diff
@@ -150,8 +150,15 @@ def unpack_header(data)
     end
 
     def get_socketpath
-      path = `i3 --get-socketpath`.chomp!
-      raise 'Unable to get i3 socketpath' unless path
+      cmd = if system('i3 --version')
+        'i3'
+      elsif system('sway --version')
+        'sway'
```

Comment:
Is `sway` fully compatible with this client?

Reply:
Sway IPC is backward compatible with i3 IPC.